elevenlabs/src/test-utils/gemini-mastra-conversation-strategy.ts

```diff
@@ -111,9 +111,49 @@ export class GeminiMastraConversationStrategy implements ConversationStrategy {
             this.messages.push(agentResponse);
         }
 
-        // Store tool calls if any
+        // Handle async tool execution pattern
         if (result.toolResults && result.toolResults.length > 0) {
             for (const toolCall of result.toolResults) {
+                // Step 1: Emit "in_progress" response and let agent react
+                const inProgressResult = {
+                    status: 'in_progress',
+                    message: 'Executing the task in the background. Result will be reported later.'
+                };
+                
+                const inProgressMessage: ServerMessage = {
+                    type: 'mcp_tool_call',
+                    mcp_tool_call: {
+                        tool_name: toolCall.payload.toolName,
+                        tool_call_id: toolCall.runId + '_in_progress',
+                        state: 'success',
+                        result: [JSON.stringify(inProgressResult)],
+                    },
+                };
+                this.messages.push(inProgressMessage);
```

Comment:
It's not enough to push these to messages. Each time, it should also be sent to the agent.

Try to find a clever way of extracting and reusing this, as there is now going to be a lot of duplication.